### DISCUSSION TITLE
Singleplayer player gets put into freecamera/spectator mode when their character is removed

### DISCUSSION THREAD

**Discussion by TheNinjaScout** [FakeFishGames/Barotrauma#15264] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Fixed in PR https://github.com/FakeFishGames/Barotrauma/pull/15265

When playing in singleplayer (without cheats), it is possible to enter freecamera by having your controlled character be removed either by status effects, by having the character's dead body be completely eaten by monsters, or by turning into a husk. Ideally, if the player is looking to play normally without cheats enabled, they should never be able to go into spectator mode if the "round" is still in progress.
https://youtu.be/gXy7ZKQSiqM

### Reproduction steps

1. Start a singleplayer campaign
2. Give your controlled crew character the husk infection
3. Wait for it to die from the infection
4. Notice how you are put into spectator mode / freecamera

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
I like this bug, i'd like to keep it and i don't think freecam at a price of dead -> despawned character is a cheat

E.g. in case of husk transformation i like to watch what my body will do and who it'll bite
And then spectate those characters being transformed and do more funny stuff

In early game when you don't have handheld sonar yet in caves or when you tell autopilot to approach target minerals and go preparing for dive
You often miss that minerals at first, swim too far and get mauled, and in those cases i like to spectate and see where those missed minerals was

Also in multiplayer you can freecam at any moment with setclientcharacter, so i don't think it's that much of a cheat


I think if you care so much, a proper solution would be to attach dead character's camera to husk body, and mb keep camera in place if body was eaten / despawned
